private/bufpkg/bufcheck/buflint/internal/buflintvalidate/validate_field.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buflintvalidate
+
+import (
+	"regexp"
+	"time"
+	"unicode/utf8"
+
+	"buf.build/gen/go/bufbuild/protovalidate/protocolbuffers/go/buf/validate"
+	"github.com/bufbuild/buf/private/pkg/protosource"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var (
+	unknown         = ""
+	httpHeaderName  = "^:?[0-9a-zA-Z!#$%&'*+-.^_|~\x60]+$"
+	httpHeaderValue = "^[^\u0000-\u0008\u000A-\u001F\u007F]*$"
+	headerString    = "^[^\u0000\u000A\u000D]*$" // For non-strict validation.
+	// Map from well-known regex to a regex pattern.
+	regexMap = map[string]*string{
+		"UNKNOWN":           &unknown,
+		"HTTP_HEADER_NAME":  &httpHeaderName,
+		"HTTP_HEADER_VALUE": &httpHeaderValue,
+		"HEADER_STRING":     &headerString,
+	}
+)
+
+// validateField is a validate Field.
+type validateField struct {
+	module   *validateModule
+	field    protosource.Field
+	location protosource.Location
+}
+
+// newValidateField returns a new validate validateField.
```

Comment:
None of these comments are public - they seem unnecessary, and inconsistent with our codebase

Reply:
resolved in [182d2e2](https://github.com/bufbuild/buf/pull/2377/commits/182d2e21541d6084e75fd345150fa990cde60d62)